From 8970b257afe9861d8971db6eb6d1e8b15ed88883 Mon Sep 17 00:00:00 2001
From: Jean-Paul Pelteret <jppelteret@gmail.com>
Date: Fri, 12 May 2017 17:11:21 +0200
Subject: [PATCH] Remove static keyword from LLVM double
 ONE_ARG_EXTERNAL_FUNCTION macro.

This fixes a segmentation fault that may arise when sequential calls are
made to the functions defined with this macro.
---
 symengine/llvm_double.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/symengine/llvm_double.cpp b/symengine/llvm_double.cpp
index 4eca24b3c..f6284499d 100644
--- a/symengine/llvm_double.cpp
+++ b/symengine/llvm_double.cpp
@@ -387,7 +387,7 @@ void LLVMDoubleVisitor::bvisit(const Log &x)
 #define ONE_ARG_EXTERNAL_FUNCTION(Class, ext)                                  \
     void LLVMDoubleVisitor::bvisit(const Class &x)                             \
     {                                                                          \
-        static llvm::Function *func = get_external_function(#ext);             \
+        llvm::Function *func = get_external_function(#ext);                    \
         auto r = builder->CreateCall(func, {apply(*x.get_arg())});             \
         r->setTailCall(true);                                                  \
         result_ = r;                                                           \
